### PR TITLE
Gui evolution xd

### PR DESCRIPTION
**Pull request by KevinMidboe**
This branch has added functionality mainly in the virtual tiles view and also some for the applications.

### CODE REVIEW

File: GATEWAY/src/pages/applications/applications.ts [hunk 1]
```diff
@@ -42,6 +42,7 @@ export class ApplicationsPage {
         this.presentLoginModal();
       } else {
         this.mqttClient.connect(val.username, val.host, val.port);
+        this.setApplications();
```
Comment (nornes) [inline GATEWAY/src/pages/applications/applications.ts:45]: Great minds think alike :P I did this in the branch I am working with too :)

File: GATEWAY/src/pages/virtual-tiles/virtual-tiles.ts [hunk 2]
```diff
@@ -10,15 +10,18 @@ import { Application, Device, UtilsService, VirtualTile } from '../../providers/
...
   selector: 'page-virtual-tiles',
   templateUrl: 'virtual-tiles.html'
 })
 export class VirtualTilesPage {
   devices: Device[];
-	applicationTitle: string;
+  applicationTitle: string;
   virtualTiles: VirtualTile[];
   activeApp: Application;
+  pairedVirtualTiles = [];
```
Comment (nornes) [inline GATEWAY/src/pages/virtual-tiles/virtual-tiles.ts:24]: Wouldn't it work to map over the tiles to check if paired or not instead? Seems redundant to store them twice.

File: GATEWAY/src/pages/virtual-tiles/virtual-tiles.ts [hunk 3]
```diff
@@ -60,7 +88,8 @@ export class VirtualTilesPage {
     const deviceRadioButtons = this.devices.map(device => {
       return {type: 'radio', name: 'deviceId', value: device.tileId, label: device.name};
     });
-    this.alertCtrl.create({
+    if (this.devices.length > 0) {
```
Comment (nornes) [inline GATEWAY/src/pages/virtual-tiles/virtual-tiles.ts:92]: Makes a lot of sense. Like it!

File: GATEWAY/src/pages/virtual-tiles/virtual-tiles.ts [hunk 4]
```diff
@@ -71,8 +100,44 @@ export class VirtualTilesPage {
...
+        message: 'No physical tiles nearby.',
+       buttons: ['Dismiss']}).present();
+    }
+  }
+  /**
+  * Called when the pair button is pushed on the view of the the
+  * the virtual tiles.
+  * @param {VirtualTile} virtualTile - the target device
+  */
+  unpairTile = (virtualTile: VirtualTile): void => {
```
Comment (nornes) [inline GATEWAY/src/pages/virtual-tiles/virtual-tiles.ts]: We do not have an api endpoint for this. You can only pair with a new one.

File: GATEWAY/src/providers/ble.service.ts [hunk 5]
```diff
@@ -1,6 +1,6 @@
 
 import { Injectable } from '@angular/core';
-import { Events } from 'ionic-angular';
+import { Events, AlertController } from 'ionic-angular';
```
Comment (nornes) [inline GATEWAY/src/providers/ble.service.ts]: why?

File: GATEWAY/src/providers/ble.service.ts [hunk 6]
```diff
@@ -30,7 +30,8 @@ export class BleService {
               private devicesService: DevicesService,
               private mqttClient: MqttClient,
   						private tilesApi: TilesApi,
-              private utils: UtilsService) {
+              private utils: UtilsService,
+              private alertCtrl: AlertController) {
```
Comment (nornes) [inline GATEWAY/src/providers/ble.service.ts]: again, why?

File: GATEWAY/src/providers/ble.service.ts [hunk 7]
```diff
@@ -1,6 +1,6 @@
```
Comment (nornes) [inline GATEWAY/src/providers/ble.service.ts]: Why was this file even added to the PR?

**Review by nornes — CHANGES_REQUESTED** [review]
Can the ble service file be removed from the PR? Doesn't seem like you have done anything except add an unused import and remove a few alerts.

**Review by eivindsandell — APPROVED** [review]
Good shit

**Review by nornes — APPROVED** [review]

**Pull request merged**